app/src/main/java/com/locoquest/app/MainActivity.kt

```diff
@@ -101,25 +107,19 @@ class MainActivity : AppCompatActivity(), OnMapReadyCallback {
                 try {
                     val credential = oneTapClient.getSignInCredentialFromIntent(data)
                     val idToken = credential.googleIdToken
-                    if (idToken != null) {
-                        // Got an ID token from Google. Use it to authenticate
-                        // with Firebase.
+                    idToken?.let{
+                        // Got an ID token from Google. Use it to authenticate with Firebase.
```

Comment:
Used let operator for save call. The code block will only be executed if the object is not null. It is better to use for mapping operation. (https://kotlinlang.org/docs/null-safety.html#checking-for-null-in-conditions)

Reply:
let: good idea.